themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,105 @@
+
+This left Zephyr again asking some questions about the best path forward:
+
+* **Should they use a single project, or multiple projects?** Zephyr had used a single Pulumi project up until now, but growth and organizational change meant they wanted to make sure they were using Pulumi in a way that best reflected their needs. What are the advantages and disadvantages of each approach?
+* **If multiple projects, how many projects?** What's the best way to determine how many projects are needed? What factors play a role in this decision?
+
+## Structuring Pulumi projects
+
+One project, or multiple projects? And if multiple projects, how many projects? These are the core questions the Zephyr needed to resolve. In the end, there are a number of key factors that determine the answers to these questions:
+
+* **Use case:** Zephyr has only a single application (so far!), but in situations where there are multiple applications then keeping those applications (and sometimes their associated infrastructure, depending on other factors) in separate projects is generally recommended.
+* **Company/department/team structure:** How a company/department/team is organized should certainly be taken into account when structuring your Pulumi projects. In Zephyr's early days with Pulumi, a single team managed everything so a single project was fine. Now with multiple teams, Zephyr will likely find using multiple projects to be more effective.
+* **The relationship between resources:** The relationships between resources---such as which resources are dependent upon other resources---is another factor that feeds into your Pulumi project structure. A "shared infrastructure layer" is often best handled in a separate project of its own, but may pull in resources that might otherwise have been kept with an application in its own project. (In some respects, this is a "counter-balance" to the first bullet point.)
+* **Resource lifecycle:** Do resources have a shared lifecycle, meaning they are always created, updated, or deleted together? (Note this is _not_ the same as a dependency on another resource!) If so, then those resources should be placed in the same project.
+* **Change rate of resources:** Some resources may change very infrequently. Consider the example of an AWS VPC and associated objects. Once it is created, it may not change for extended periods of time, if ever. On the other hand, some resources may change frequently (in other words, these resources have a high change rate). In a single project, combining resources that have a high change rate with resources that have a low change rate may be suboptimal.
```

Comment:
If we mean "blast radius", we should use that term. (And blast radius is definitely a concern with repo structure.)

Reply:
We don't mean blast radius here; that's mentioned later in the article.